File: wiz-shop-java/src/main/java/com/example/wizshopjava/domain/member/entity/Member.java
package com.example.wizshopjava.domain.member.entity;

import com.example.wizshopjava.common.BaseEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Member extends BaseEntity {

	@Id @GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "member_id")
	private Long id;

	private String email;
	private String password;
	private String nickname;

	@Builder
	public Member(Long id, String email, String password, String nickname) {
		this.id = id;
		this.email = email;
		this.password = password;
		this.nickname = nickname;
	}
}

File: wiz-shop/src/main/kotlin/com/example/wizshop/domain/member/Member.kt
package com.example.wizshop.domain.member

import com.example.wizshop.common.BaseEntity
import jakarta.persistence.*
import org.springframework.security.crypto.password.PasswordEncoder

@Entity
@Table(name = "member")
class Member(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    var id: Long? = null,

    val email: String,
    var password: String,
    var nickname: String

): BaseEntity() {

    fun encodePassword(encoder: PasswordEncoder) {
        this.password = encoder.encode(password)
    }
}